src/packageurl/contrib/purl2url.py

```diff
@@ -0,0 +1,151 @@
+    namespace = data.get('namespace')
+    if not (name and namespace):
+        return
+
+    url = 'https://bitbucket.org/{namespace}/{name}'.format(
+        namespace=namespace, name=name
+    )
+    version = data.get('version')
+    if version:
+        url = '{url}/src/{version}'.format(url=url, version=version)
+
+    subpath = data.get('subpath')
+    if subpath:
+        url = '{url}/{subpath}'.format(url=url, subpath=subpath)
+
+    return url
+
+
+def build_cargo_download_url(**data):
+    """
+    Return a Cargo URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    version = data.get('version')
+    if not (name and version):
+        return
+
+    return 'https://crates.io/api/v1/crates/{name}/{version}/download'.format(
+        name=name, version=version
+    )
+
+
+def build_github_homepage_url(**data):
+    """
+    Return a GitHub URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    namespace = data.get('namespace')
+    if not (name and namespace):
+        return
+
+    url = 'https://github.com/{namespace}/{name}'.format(namespace=namespace, name=name)
+
+    version = data.get('version')
+    if version:
+        url = '{url}/tree/{version}'.format(url=url, version=version)
+
+    subpath = data.get('subpath')
+    if subpath:
+        url = '{url}/{subpath}'.format(url=url, subpath=subpath)
+
+    return url
+
+
+def build_gitlab_homepage_url(**data):
+    """
+    Return a GitLab URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    namespace = data.get('namespace')
+
+    if not (name and namespace):
+        return
+
+    url = 'https://gitlab.com/{namespace}/{name}'.format(namespace=namespace, name=name)
+
+    version = data.get('version')
+    if version:
+        url = '{url}/-/tree/{version}'.format(url=url, version=version)
+
+    subpath = data.get('subpath')
+    if subpath:
+        url = '{url}/{subpath}'.format(url=url, subpath=subpath)
+
+    return url
+
+
+def build_gem_download_url(**data):
+    """
+    Return a RubyGem URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    version = data.get('version')
+
+    if not (name and version):
+        return
+
+    return 'https://rubygems.org/downloads/{name}-{version}.gem'.format(
+        name=name, version=version
+    )
+
+
+def purl2url(purl):
+    """
+    Return a valid URL `url` according to the type of given PURL `purl`
+    """
+    purl_data = PackageURL.from_string(purl)
+    
+    converter = {
+        'bitbucket': build_bitbucket_homepage_url,
+        'cargo': build_cargo_download_url,
+        'rubygems': build_gem_download_url,
+        'github': build_github_homepage_url,
+        'gitlab': build_gitlab_homepage_url,
+    }
+
+    type = purl_data.type
+    if not type in converter:
+        return
+
+    url = converter[type](
+        name=purl_data.name,
```

Comment:
Related to my comments above for the converter funtions, since you are passing concret arg names... why using **data in each of the signatures?
You should instead use proper arg names in each of you functions (e.g. name, namespace, version, etc...)